### PR TITLE
API Provides User the Functionality to Login

### PR DESCRIPTION
**Pull request by blake-futchi**
# [PT Story](https://www.pivotaltracker.com/story/show/171467555)

```
As a restaurant API
In order to welcome previous users to continue their order
I would like to give them the ability to login
```

## Changes proposed in this pull request:
* Adds unit test for User model, and request spec for User authentication
* Installs and setups Devise Token Auth
## What I have learned working on this feature:
* Modifying User model from Devise Token Auth with new DB column, and configuring permitted params 
* Refreshed knowledge about implementing and testing Devise Token Auth
## Packages/Gems added
- devise_token_auth

### CODE REVIEW

**Comment by tochman** [comment]
I see the same commits that we see in another PR as part of this one. Wait until we merge the products PR before submitting this one. Generally, you NEED to branch off from development and NOT another feature branch (as you did here)

**Comment by blake-futchi** [comment]
Thanks Thomas for the feedback, as mentioned in the slack conversation we were in a bottleneck and had some base issues on our dev branch we solved in the feature branch.

**Review by oliverochman — APPROVED** [review]

**Pull request merged**
